core/src/test/java/org/infinispan/persistence/file/SingleFileStoreStressTest.java

```diff
@@ -191,65 +188,53 @@ public void testFileTruncation() throws ExecutionException, InterruptedException
       final int TEST_DURATION_SECONDS = 2;
 
       Cache<String, String> cache = cacheManager.getCache(CACHE_NAME);
-      PersistenceManager persistenceManager = TestingUtil.extractComponent(cache, PersistenceManager.class);
-      final SingleFileStore store = persistenceManager.getStores(SingleFileStore.class).iterator().next();
+      final SingleFileStore store = TestingUtil.getFirstWriter(cache);
       final StreamingMarshaller marshaller = TestingUtil.extractComponentRegistry(cache).getCacheMarshaller();
       assertEquals(0, store.size());
 
       // Write a few entries into the cache
-      final List<String> keys = populateStore(NUM_KEYS, 0, store, marshaller);
+      final List<String> keys = populateStore(NUM_KEYS, 2, store, marshaller);
 
       // Do some reading/writing entries with random size
       final CountDownLatch stopLatch = new CountDownLatch(1);
       Future[] writeFutures = new Future[NUM_WRITER_THREADS];
-      for (int i = 0; i < NUM_WRITER_THREADS; i++) {
+      for (int i = 0; i < NUM_WRITER_THREADS; i++)
          writeFutures[i] = fork(stopOnException(new WriteTask(store, marshaller, keys, stopLatch), stopLatch));
-      }
 
       Future[] readFutures = new Future[NUM_READER_THREADS];
-      for (int i = 0; i < NUM_READER_THREADS; i++) {
+      for (int i = 0; i < NUM_READER_THREADS; i++)
          readFutures[i] = fork(stopOnException(new ReadTask(store, keys, false, stopLatch), stopLatch));
-      }
 
       stopLatch.await(TEST_DURATION_SECONDS, SECONDS);
       stopLatch.countDown();
 
-      for (int i = 0; i < NUM_WRITER_THREADS; i++) {
+      for (int i = 0; i < NUM_WRITER_THREADS; i++)
          writeFutures[i].get();
-      }
-      for (int i = 0; i < NUM_READER_THREADS; i++) {
+
+      for (int i = 0; i < NUM_READER_THREADS; i++)
          readFutures[i].get();
-      }
 
       File file = new File(location, CACHE_NAME + ".dat");
       long length1 = file.length();
 
-      ExecutorService executor = Executors.newSingleThreadExecutor(getTestThreadFactory("Purge"));
-      store.purge(executor, null);
-      // Give some time for the purge thread to finish
-      MILLISECONDS.sleep(200);
+      store.purge(null, null);
       long length2 = file.length();
 
       // Again write entries with smaller size
       populateStore(NUM_KEYS, 0, store, marshaller);
-
-      store.purge(executor, null);
-      // Give some time for the purge thread to finish
-      MILLISECONDS.sleep(200);
+      store.purge(null, null);
       long length3 = file.length();
 
-      executor.shutdown();
-
       // Verify that file size decreases and the file shrinks
       assertTrue(length2 <= length1);
-      assertTrue(length3 < length2);
+      assertTrue(String.format("Length2=%d, Length3=%d", length2, length3), length3 <= length2);
```

Comment:
Looking at the test some more, it's sleeping for 200ms, but the `SingleFileStore.purge()` actually does everything on the caller thread. So the test could pass a `WithinThreadExecutor` and skip the wait.

I also suggest testing `store.getFileSize()` before `file.size()`, just in case there's some caching going on in `File.size()`. I'd go even further and use `TestingUtil.extractField` to print the used+free entries if the `store.getFileSize()` check fails.